package-res/resources/web/pentaho/lang/ActionResult.js

```diff
@@ -0,0 +1,164 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+define([
+  "./Base",
+  "./UserError",
+  "../util/error"
+], function(Base, UserError, utilError) {
+  "use strict";
+
+  /**
+   * @classDesc An `ActionResult` represents the result from performing an _action_.
+   * 
+   * ### `ActionResult` Value
+   * 
+   * When successful an _action_ can return a value that represents its final state or `undefined`, for example,
+   * if you do a google search using an {@link pentaho.visual.base.model#executeAction|executeAction}, there is
+   * no value to be returned.
+   *
+   * ### `ActionResult` Error
+   *
+   * When an _action_ is unsuccessful it will return an error or a string with the reason why it was rejected.
```

Comment:
When an _action_ is unsuccessful, it will return an error or a string with the reason why it was rejected.
This can happen when an event is canceled, or the core components of that _action_ were invalid.